packages/wrangler/src/deployment-bundle/esbuild-plugins/log-build-output.ts

```diff
@@ -0,0 +1,46 @@
+import { logBuildFailure, logBuildWarnings } from "../../logger";
+import { rewriteNodeCompatBuildFailure } from "../build-failures";
+import type { NodeJSCompatMode } from "../node-compat";
+import type { Plugin } from "esbuild";
+
+/**
+ * Log esbuild warnings and errors
+ */
+export function logBuildOutput(
+	nodejsCompatMode: NodeJSCompatMode | undefined,
+	onStart?: () => void,
+	updateBundle?: () => void
+): Plugin {
+	let bundled = false;
+
+	return {
+		name: "log-build-output",
+		setup(b) {
+			b.onStart(() => {
+				onStart?.();
+			});
+			b.onEnd(async (result) => {
+				const errors = result.errors;
```

Comment:
Nit: Destructure the function arg instead perhaps?

Reply:
https://github.com/cloudflare/workers-sdk/pull/6566/commits/0d6eeb9fa8f6831cad98d3934ff38faf0c5a3b0a